queries.sql

```diff
@@ -1,3 +1,199 @@
+WHERE species IS NULL;
+
+SELECT * FROM animals;
+
+ROLLBACK;
+BEGIN TRANSACTION;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+SELECT * FROM animals;
+
+COMMIT;
+
+-- Begin Second Transaction
+
+BEGIN TRANSACTION;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+-- Begin Third Transaction
+BEGIN TRANSACTION;
+DELETE FROM animals 
+WHERE date_of_birth > 'Jan 1 2022';
+SAVEPOINT savepoint_weight;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO savepoint_weight;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+COMMIT;
+
+-- Begin forth transaction
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*)
+FROM animals
+WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg)
+FROM animals;
+
+SELECT neutered, COUNT(*) AS escape_count FROM animals
+WHERE escape_attempts > 0
+GROUP BY neutered
+ORDER BY escape_count DESC
+LIMIT 1;
+
+SELECT species,
+MIN(weight_kg) AS min_weight,
+MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+SELECT species, AVG(escape_attempts) AS 
+avg_escape_attempts 
+FROM animals
+WHERE date_of_birth BETWEEN 'Jan 1 1990' AND 'Dec 31 2000'
+GROUP BY species;
+
+-- Query multipl TABLE
+-- What animals belong to Melody Pond?
+SELECT A.name, O.full_name
+FROM animals A
+JOIN owners O ON A.owner_id = O.id
+WHERE O.full_name = 'Melody Pond';
+
+-- List of all animals that are pokemon (their type is Pokemon).
+SELECT A.name
+FROM animals A
+JOIN species S ON A.species_id = S.id
+WHERE S.name = 'Pokemon';
+
+-- List all owners and their animals, including those that don't own any animal.
+SELECT A.name, O.full_name
+FROM owners O
+LEFT JOIN animals A ON O.id = A.owner_id;
+
+-- How many animals are there per species?
+SELECT S.name AS species_name, COUNT(A.id) AS animal_count
+FROM species S
+LEFT JOIN animals A ON S.id = A.species_id
+GROUP BY S.name;
+
+-- List all Digimon owned by Jennifer Orwell.
+SELECT A.name AS digimon_name
+FROM animals A
+JOIN species S ON A.species_id = S.id
+JOIN owners O ON A.owner_id = O.id
+WHERE O.full_name = 'Jennifer Orwell'
+AND S.name = 'Digimon';
+
+-- List all animals owned by Dean Winchester that haven't tried to escape.
+SELECT A.NAME
+FROM animals A
+JOIN owners O ON A.owner_id = O.id
+WHERE O.full_name = 'Dean Winchester'
+AND A.escape_attempts = 0;
+
+-- Who owns the most animals?
+SELECT O.full_name AS owner_name, COUNT(A.id) AS animal_count
+FROM owners O
+LEFT JOIN animals A ON O.id = A.owner_id
+GROUP BY O.full_name
+ORDER BY COUNT(A.id) DESC
+LIMIT 1;
+
+-- Join Tables for visits
+
+SELECT A.name AS last_animal_seen 
+FROM animals A
+JOIN visits V ON V.animal_id = A.id
+JOIN vets VE ON V.vet_id = VE.id
+WHERE VE.name = 'William Tatchar'
+ORDER BY V.date_of_visit DESC
+LIMIT 1;
+
+SELECT COUNT(DISTINCT A.id)
+FROM animals A
+JOIN visits V ON A.id = V.animal_id
+JOIN vets VE ON V.vet_id = VE.id
+WHERE VE.name = 'Stephanie Mender';
+
+
```

Comment:
- You have done great so far :+1: :+1: But there are a few changes you need to make before moving to the next milestone. :+1: :+1: 

- [ ] From the screenshot of your query result, the first two queries `Who was the last animal seen by William Tatcher?` and `How many different animals did Stephanie Mendez see?` returned `0` as the result. This is not correct. THe result of the first query is meant to be ``Blossom`` while the second one is meant to be a count of `4`, I'll suggest you take a look at your query or the data you added into the database. :+1: :100: :smile:

Reply:
Hi @V-Blaze,

Thanks so much for your constructive and detailed review.
I appreciate!